### ISSUE TITLE
Seg fault in src/app/util/attribute-storage.cpp

### ISSUE THREAD

**Issue by latch-danielkneip** [project-chip/connectedhomeip#22272]
#### Problem
We’ve run into a seg fault in src/app/util/attribute-storage.cpp when emAfEndpoints[ep].endpointType is 0 and emberAfFindClusterInType() is called from emberAfFindClusterIncludingDisabledEndpoints(). Adding  emAfEndpoints[ep].endpointType != nullptr before the call allows us to function normally. However, I’m concerned there may be a deeper issue.

The seg fault occurs after issuing the following command to dynamic endpoint 4 that represents a bridged dimmer:
```
./chip-tool onoff toggle 1 4
```

 #### Proposed Solution
```
diff --git a/src/app/util/attribute-storage.cpp b/src/app/util/attribute-storage.cpp
index 7e8bad312..7d3124d30 100644
--- a/src/app/util/attribute-storage.cpp
+++ b/src/app/util/attribute-storage.cpp
@@ -796,7 +796,7 @@ const EmberAfCluster * emberAfFindClusterIncludingDisabledEndpoints(EndpointId e
                                                                     EmberAfClusterMask mask)
 {
     uint16_t ep = emberAfIndexFromEndpointIncludingDisabledEndpoints(endpoint);
-    if (ep < MAX_ENDPOINT_COUNT)
+    if (ep < MAX_ENDPOINT_COUNT && emAfEndpoints[ep].endpointType != nullptr)
     {
         return emberAfFindClusterInType(emAfEndpoints[ep].endpointType, clusterId, mask);
     }
```

**Comment by bzbarsky-apple** [comment]
If we don't fix this for 1.0, I expect we will have common crashes on any app that allocates some dynamic endpoints but does not actually have endpoints defined for all of them (e.g. any bridge that is not maxed out in terms of the number of devices it's bridging).

**Comment by andy31415** [comment]
Master patch acceptable: segfault fix.
